ols/src/cache/redis_cache.py

```diff
@@ -74,10 +78,12 @@ def insert_or_append(self, user_id: str, conversation_id: str, value: str) -> No
                 memory is higher than maxmemory.
         """
         key = super().construct_key(user_id, conversation_id)
-
         old_value = self.get(user_id, conversation_id)
         with self._lock:
             if old_value:
-                self.redis_client.set(key, old_value + "\n" + value)
+                old_value.append(value)
+                self.redis_client.set(key, old_value)
             else:
-                self.redis_client.set(key, value)
+                values = []
+                values.append(value)
+                self.redis_client.set(key, values)
```

Comment:
shall we care about whether redis has saved our conversations?
If missing a message in the conversation, will it affect greatly the quality of the answer to user's questions?

Reply:
@raptorsun Agree , but  this is how it was , can we open a bug to address this ?

Reply:
> Agree , but this is how it was , can we open a bug to address this ?

the point of OLS-110 is to do the "real implementation" 

There are some things where it's reasonable to queue it up for a follow up PR, but this doesn't seem like one of those cases.  We need proper error handling+logging when we implement new functionality.